### PR TITLE
feat: Add OAuth ESM + Vite + JS sample

### PR DESCRIPTION
**Pull request by lboyd93**
This PR adds an OAuth example using Vite and JS with ESM.

@andygup would you please review? Thanks!

### CODE REVIEW

File: esm-samples/jsapi-oauth/style.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import "https://js.arcgis.com/4.28.10/@arcgis/core/assets/esri/themes/light/main.css";
```
Comment (andygup) [inline esm-samples/jsapi-oauth/style.css:1]:
```suggestion
@import "https://js.arcgis.com/4.29/@arcgis/core/assets/esri/themes/light/main.css";
```

File: esm-samples/jsapi-oauth/main.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+});
+
+// Add the OAuthInfo to the IdentityManager.
+esriId.registerOAuthInfos([info]);
+
+// Call the checkSignIn function to see if the user is already signed in.
+checkSignIn();
+
+// Function to sign in or out of the portal
+function signInOrOut() {
```
Comment (andygup) [inline esm-samples/jsapi-oauth/main.js:32]: Will this still work if you convert the classic functions to arrow functions? That's the pattern used in the other samples.

File: esm-samples/jsapi-oauth/main.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        .then(() => {
+          // Once a credential is returned from the promise, check the
+          // sign in status.
+          checkSignIn();
+        });
+    });
+}
+
+// Function to check the current sign in status and display username if signed in.
+function checkSignIn() {
```
Comment (andygup) [inline esm-samples/jsapi-oauth/main.js:57]: Ditto on converting to an arrow function...

File: esm-samples/jsapi-oauth/oauth-callback.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
```
Comment (andygup) [inline esm-samples/jsapi-oauth/oauth-callback.html:1]: @dasa seems like with the addition of this ESM sample, maybe we can combine and remove our current oauth callback example into this one? https://github.com/Esri/jsapi-resources/tree/main/oauth

File: esm-samples/jsapi-oauth/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview"
+  },
+  "devDependencies": {
+    "vite": "^5.1.0"
+  },
+  "dependencies": {
+    "@arcgis/core": "^4.28.10",
```
Comment (andygup) [inline esm-samples/jsapi-oauth/package.json:15]:
```suggestion
    "@arcgis/core": "~4.29.0",
```

File: esm-samples/jsapi-oauth/package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "install": "^0.13.0",
```
Comment (andygup) [inline esm-samples/jsapi-oauth/package.json:16]: Can we remove the `install` and `npm` dependencies from the package.json?

File: esm-samples/jsapi-oauth/index.html [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Access ArcGIS Online using OAuth 2.0 - Vite + JS</title>
```
Comment (andygup) [inline esm-samples/jsapi-oauth/index.html:7]:
```suggestion
    <title>Access ArcGIS Online using OAuth 2.0 - Vite + ArcGIS</title>
```

**Review by andygup — CHANGES_REQUESTED** [review]
Thanks for pulling this together @lboyd93! I just have a handful of small change requests.

File: esm-samples/jsapi-oauth/index.html (hunk 7 above)
Comment (andygup) [reply esm-samples/jsapi-oauth/index.html:7]: Another option for the `<title>` might be to use `OAuth + Vite + ArcGIS`.

File: esm-samples/jsapi-oauth/oauth-callback.html (hunk 4 above)
Comment (dasa) [reply esm-samples/jsapi-oauth/oauth-callback.html:1]: OK, but I don't think of that as an example callback page, but more of as the canonical callback page source that users can change if they want to. We also link to it from 3.x and 4.x doc, so might be a little odd of have it in only in an example app.

Comment (andygup) [reply esm-samples/jsapi-oauth/oauth-callback.html:1]: Ah yes that makes sense, I forgot about the SDK links to that directory and that we suggest users can download the callback file and host it on their server. In that case, disregard my previous comment and let's keep the `/oauth` directory, as-is.

File: esm-samples/jsapi-oauth/main.js (hunk 2 above)
Comment (lboyd93) [reply esm-samples/jsapi-oauth/main.js:32]: It should! I'll update and test it again.

**Comment by lboyd93** [comment]
> Thanks for pulling this together @lboyd93! I just have a handful of small change requests.

@andygup changes made! Feel free to re-review or merge if all is good. 👍

**Review by andygup — APPROVED** [review]
LGTM!

**Pull request merged**